README.md

```diff
@@ -21,68 +27,147 @@ After installing and configuring those requirements, you can download [docker-co
 
-Once it is cloned you can easily open a shell in the docker container to
-open an development environment.
+**Question** defines a single input of a form whereas a question has different implementation types defining how a question should be represented and validated.
 
-```bash
-# needed for permission handling
-# only needs to be run once
-echo UID=$UID > .env
-# open shell
-docker-compose run --rm caluma bash
-```
+**Document** is a specific instance of a form being a container of answers.
 
-### Testing
+**Answer** containing user input of a specific question. There are different implementations of answer covering different data types.
 
-Once you have shelled in docker container as described above
-you can use common python tooling for formatting, linting, testing
-etc.
+#### Workflow entities
 
-```bash
-# linting
-flake8
-# format code
-black .
-# running tests
-pytest
-# create migrations
-./manage.py makemigrations
-```
+Naming and concept of workflow follows concept of [Workflow Patterns Initiative](http://www.workflowpatterns.com/).
 
-### Install new requirements
+**Workflow** defines a structure of a business process. It uses forward moving flows referencing tasks to model it.
 
-In case you needed to add new requirements you simply need to build the docker container
-again for those to be installed and re-open shell.
+**Task** is a definition of work being part of a workflow.
+
+**Flow** is a definition on what task resp. tasks follows another.
+
+**Case** is a specific instance of a workflow.
+
+**WorkItem** is a specific instance of a task a resource has to work off.
+
+
+### Configuration
+
+Caluma is a [12factor app](https://12factor.net/) so it is configurable per environment variables.
+Different environment variable types are explained at [django-environ](https://github.com/joke2k/django-environ#supported-types)
+
+#### Common
+
+A list of configuration options which you might need to configure to get Caluma started in your environment.
+
+* `SECRET_KEY`: A secret key used for cryptography. See [more](https://docs.djangoproject.com/en/2.1/ref/settings/#std:setting-SECRET_KEY)
+* `ALLOWED_HOSTS`: A comma separated list of strings representing host/domans Caluma will be served on. See [more](https://docs.djangoproject.com/en/2.1/ref/settings/#allowed-hosts)
+* `DATABASE_NAME`: Name of database to use (default: caluma)
+* `DATABASE_USER`: username to use when connecting to the database (default: caluma)
+* `DATABASE_PASSWORD`: password to use when connecting to database
+* `DATABASE_HOST`: host to use when connecting to Postgres database (default: localhost)
+* `DATABASE_PORT`: port to use when connecting to Postgres database (default: 5432)
+* `LANGUAGE_CODE`: default language defined as fall back (default: en)
+
+#### OpenID Connect
+
+When you want to connect Caluma to your [IAM](https://en.wikipedia.org/wiki/Identity_management) supporting OpenID connect you need following settings.
```

Comment:
What if I don't? Do we support a standalone authorisation/authentication scheme?

Reply:
good point. Rewrote it that for authentication an IAM is required. If not available suggesting Keycloak.